DataProcessing/CryptoCoarseFundamentalUniverseDataDownloader.cs

```diff
@@ -0,0 +1,283 @@
+                                continue;
+                            }
+
+                            var conversionRate = decimal.Parse(coarseByDate[refCurrency].Split(",").First(), NumberStyles.Any, CultureInfo.InvariantCulture);
+                            conversionRate = refCurrency.Substring(0, 3) == "USD" || 
+                                refCurrency.Substring(0, 4) == "BUSD" ? 
+                                1m / conversionRate : 
+                                conversionRate;
+                            var baseDollarVol = decimal.Parse(coarseByDate[dataTicker].Split(",")[2], NumberStyles.Any, CultureInfo.InvariantCulture);
+                            usdDollorVol = $"{baseDollarVol * conversionRate * rateToUSD}";
+                        }
+                        else
+                        {
+                            usdDollorVol = $"{coarseByDate[dataTicker].Split(",")[2]}";
+                        }
+
+                        dataByDate[fileName][dataTicker] = $"{dataByDate[fileName][dataTicker]}{usdDollorVol}";
+                    }
+
+                    // Save to file
+                    var fileContent = dataByDate[fileName].Select(x => 
+                        {
+                            var ticker = x.Key;
+                            var sid = SecurityIdentifier.GenerateCrypto(ticker, _market);
+                            return $"{sid},{ticker},{x.Value}";
+                        })
+                        .ToList();
+                    var finalPath = Path.Combine(_destinationFolder, $"{fileName}.csv");
+                    var finalFileExists = File.Exists(finalPath);
+
+                    var lines = new HashSet<string>(fileContent);
+                    if (finalFileExists)
+                    {
+                        foreach (var line in File.ReadAllLines(finalPath))
+                        {
+                            lines.Add(line);
+                        }
+                    }
+
+                    var finalLines = lines.OrderBy(x => x.Split(',').First()).ToList();
+
+                    var tempPath = Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.tmp");
+                    File.WriteAllLines(tempPath, finalLines);
+                    var tempFilePath = new FileInfo(tempPath);
+                    tempFilePath.MoveTo(finalPath, true);
+                }
+            }
+            catch (Exception e)
+            {
+                Log.Error(e);
+                return false;
+            }
+
+            Log.Trace($"CryptoCoarseFundamentalUniverseDataConverter.ConvertToUniverseFile(): Finished Processing");
+            return true;
+        }
+
+        /// <summary>
+        /// Helper method to read lines in a online file.
+        /// </summary>
+        /// <param name="link">The link of online file to be read</param>
+        public IEnumerable<string> ReadOnlineDocumentLines(string link)
+        {
+            WebClient client = new WebClient();
```

Comment:
`WebClient` shouldn't be use anymore, there's an extension method in Lean which provides similar functionality